spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -20,26 +20,52 @@ def foobar_action
 end
 
 RSpec.describe Hq::PartnersController, type: :controller do
-  before :each do
-    sign_in instance_double(AdminUser)
-    @partner = build_stubbed :partner
-    expect(Partner).to receive(:find).and_return(@partner)
-  end
 
-  specify 'editing renders the edit view' do
-    get :edit, id: @partner.id
-    expect(response).to render_template 'edit'
-  end
+  context '#new and #create' do
+    
+    before :each do
+      sign_in instance_double(AdminUser)
+      @partner = FactoryGirl.build(:partner)
+    end
+    
+    specify 'successful create redirects to the show view' do
+      post :create, partner: @partner.attributes
+      @partner = Partner.find_by_name(@partner[:name])
+      expect(response).to redirect_to hq_partner_path(@partner)
+    end
```

Comment:
This test touches the DB 2 times (one for `create` and one for `find_by_id`) and I don't like it.
We were fighting this last night to make the test work without using the db but no luck.
If anybody has an idea how to do this properly please share.
(i didn't sleep well because of it :) )